components/list/list-item-nav-mixin.js

```diff
@@ -106,6 +127,24 @@ export const ListItemNavMixin = superclass => class extends ListItemLinkMixin(su
 	dispatchSetChildCurrentEvent(val) {
 		/** @ignore */
 		this.dispatchEvent(new CustomEvent('d2l-list-item-nav-set-child-current', { bubbles: true, composed: true, detail: { value: val } }));
+
+		if (!val) return;
+		requestAnimationFrame(() => {
+			if (this._hasNestedList) {
+				const firstChild = this.querySelector('[first]');
+				if (firstChild) firstChild._hasCurrentParent = true;
+			}
+
+			const prevSibling = this._getPreviousListItemSibling();
+			if (prevSibling) {
+				prevSibling._nextSiblingCurrent = true;
+
+				if (prevSibling._hasNestedList) {
+					const lastChild = prevSibling.querySelector('[last]');
+					if (lastChild) lastChild._nextSiblingCurrent = true;
+				}
+			}
+		});
```

Comment:
When an item is current, we want both the top and bottom add buttons to blend nicely into the current item's border. The difficulty is that when a list is nested and/or an item has a nested list the add or add-top might be part of the adjacent item. This logic is to add `_hasCurrentParent` and/or `_nextSiblingCurrent` attributes as needed.